RIOT-OS/RIOT
# gnrc_sixlowpan_frag: initial import of Selective Fragment Recovery

### Contribution description
This provides an implementation for [6LoWPAN Selective Fragment Recovery](https://tools.ietf.org/html/draft-ietf-6lo-fragment-recovery-07) (SFR), an alternative fragmentation protocol for 6LoWPAN, which allows for end-to-end recovery and congestion control (currently not implemented) for fragmented IPv6 datagrams.



### Testing procedure
There is still a lot of more in-depth testing required including a dedicated test application (this is why is it still WIP), but you can compile `examples/gnrc_networking` with SFR already for a 6LoWPAN-capable board:

```
USEMODULE="gnrc_sixlowpan_frag_sfr" BOARD=samr21-xpro make -C examples/gnrc_networking flash term
```

and then ping another node also shipping with `gnrc_sixlowpan_frag_sfr`:

```
ping6 -s 1232 fe80::7b65:822:8693:9d5a
```

~~*[edit]: after posting this PR, I noticed, that there is a leak in the packet buffer after this... All the more reason why this is still WIP[/edit]*~~

There are also tests that should pass on a board of your choice:

```sh
BOARD=... make -C tests/gnrc_sixlowpan_frag_sfr -j flash term
```




### Issues/PRs references
Requires ~~#12220~~ (merged), ~~#12559~~ (merged), ~~#12629~~, ~~#12845~~, and ~~#12848~~ (merged):

![Route to 6LoWPAN fragment forwarding](http://page.mi.fu-berlin.de/mlenders/sixlo_minfwd.svg)
